### ISSUE TITLE
BPD abnormally high

### ISSUE THREAD

**Issue by royd1990** [chrischute/glow#6]
Hi,

I am taking the code from the repo as-is and running it. The only difference is I am running it a different PyTorch version (1.6.0).

For some test_runs (for e.g. 10, 20 epochs), the BPD score is abnormally high, 

**50000/50000 [16:59<00:00, 49.03it/s, bpd=9.49e+11, lr=0.0003, nll=2.02e+15]**

For 2/3 runs even after a 50 epochs I am getting such scores.
Is there something wrong that I am doing during the training or the pytorch version is affecting the results so much?

**Comment by chrischute** [comment]
It's probably caused by changing the PyTorch version. This is why the `environment.yml` file includes version numbers.

Just run `conda env create -f environment.yml` to set up your environment with the right package versions.

**Comment by DiegoOrtego** [comment]
Hi @royd1990,

I had the same issue, and apparently there are some changes in the PyTorch scheduler in newer versions. So, changing "scheduler.step(global_step)-->scheduler.step()" in train.py solved the problem in my case, as suggested by the warning that pops up when running train.py.

**Comment by swyoon** [comment]
Dear @DiegoOrtego

I had a similar problem of exploding BTD, and your solution does fix the problem.

Thank you very much!

**Comment by tayalkshitij** [comment]
Dear @DiegoOrtego, @swyoon 

I also have the same issue of abnormally high BPD and was resolved by changing changing "`scheduler.step(global_step)-->scheduler.step()'`. However, the BPD is now decreasing at a very minimal rate. In the original implementation, for around 10 epochs, the BPD should be  3.61, but for me, after the change in the scheduler, even after 25 epochs, my BPD is 3.92. Any suggestion for reducing BPD faster? 

Thank you very much!
